addons/mail/models/mail_thread.py

```diff
@@ -2268,7 +2268,10 @@ def _process_attachments_for_post(self, attachments, attachment_ids, message_val
                         node.set('src', f'/web/image/{att_id}?access_token={token}')
                         postprocessed = True
                 if postprocessed:
+                    body_is_markup = isinstance(body, Markup)
                     return_values['body'] = lxml.html.tostring(root, pretty_print=False, encoding='unicode')
+                    if body_is_markup:
+                        return_values['body'] = Markup(return_values['body'])
```

Comment:
for master:
```suggestion
                    assert isinstance(body, Markup)
                    return_values['body'] = Markup(lxml.html.tostring(root, pretty_print=False, encoding='unicode'))
```
or
```suggestion
                    body = escape(body)  # L2182
                    ...
                    return_values['body'] = Markup(lxml.html.tostring(root, pretty_print=False, encoding='unicode'))
```
Every call to `_process_attachments_for_post` should be using a markup in the body

Reply:
Hello @mart-e,

Why did you write this "for master"?
And if we want markup in the body for each call
So, What is the need for step with assert in 1st case and escape in 2nd case?
What is the problem if we write this line alone?
`return_values['body'] = markup(lxml.html.tostring(root, pretty_print=False, encoding='unicode'))`
I am a little confused

Thanks :blush:

Reply:
> Why did you write this "for master"?

Because it's quite risky in stable, you prefer badly escaped content than a crash.

>  So, What is the need for step with assert in 1st case and escape in 2nd case?

Two ways to be sure  `body` is always a markup object. In the first you have a crash if body is not in the right format. In the second, you escape if it was a string.

>  What is the problem if we write this line alone?

If we are certain than body is always a markup object, it's fine to do so. But if a public method pass a string to the method, you just found a way to turn untrusted text into markup. So you need to check it somehow.

Reply:
So, should I re-target it to master?

And saas-16.3, saas-16.4, and 17.0 should be kept as it is?